### PR TITLE
Changed the torch.data.utils.random_split() function 

### PR DESCRIPTION
**Pull request by romanovacca**
Change to take in a list of decimals resenting a percentage in how you want the dataset to be split, instead of putting in the lengths of the splits you want.

**edit**: added the issue number: #[11980](https://github.com/pytorch/pytorch/issues/11980)

### CODE REVIEW

**Review by apaszke — CHANGES_REQUESTED** [review]
Converting percentages to lengths is just `[p * len(dataset) for p in percentages]`, so I don't think it's worth sacrifising backwards compatibility for it (and it doesn't seem like "the right thing to do" either).

**Comment by romanovacca** [comment]
@apaszke I added the original issue where I explain why requested this PR. 
Also, would it be an option to maybe adjust the PR code so that it's backward-compatible, by accepting lengths AND percentages?

**Comment by romanovacca** [comment]
@apaszke?

**Comment by fmassa** [comment]
Can't you add a normalization step beforehand, so that previous behavior is maintained, while also supporting probabilities?

**Comment by MiniXC** [comment]
I don't know about it not being "the right thing to do". E.g. you can give [train_test_split](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html) a ``test_size`` in percent which is very intuitive.

Also it's not just ``[p * len(dataset) for p in percentages]`` because you need to cast to ``int``, and then think about rounding.

**Comment by gchanan** [comment]
> Also it's not just [p * len(dataset) for p in percentages] because you need to cast to int, and then think about rounding.

Right, but there's no getting around thinking about rounding.  Why do we know better than the user what to do?
